cmd/ops_agent_uap_plugin/service_linux.go

```diff
@@ -72,14 +72,17 @@ func (ps *OpsAgentPluginServer) Apply(ctx context.Context, msg *pb.ApplyRequest)
 // Until plugin receives Start request plugin is expected to be not functioning
 // and just listening on the address handed off waiting for the request.
 func (ps *OpsAgentPluginServer) Start(ctx context.Context, msg *pb.StartRequest) (*pb.StartResponse, error) {
+	ps.mu.Lock()
 	if ps.cancel != nil {
 		log.Printf("The Ops Agent plugin is started already, skipping the current request")
+		ps.mu.Unlock()
```

Comment:
Why to use `defer ps.mu.Unlock()` the same as in `Stop` ?

Reply:
Because the mutex unlock on L78 should not wait until the `Start()` method is about to return. `Start()` is a long function, unlock `Stop()` and `GetStatus()`

The only thing `Stop()` and `GetStatus()` do is to access the field `cancel`, that's not the case with `Start()`. `Start` does additional stuff like: run config generator, start up subagents etc.